packages/cursorless-engine/src/processTargets/modifiers/scopeHandlers/BoundedNonWhitespaceSequenceScopeHandler.ts

```diff
@@ -0,0 +1,122 @@
+import {
+  Direction,
+  Position,
+  ScopeType,
+  TextEditor,
+  type Range,
+} from "@cursorless/common";
+import type { Target } from "../../../typings/target.types";
+import { TokenTarget } from "../../targets";
+import { BaseScopeHandler } from "./BaseScopeHandler";
+import { compareTargetScopes } from "./compareTargetScopes";
+import type { TargetScope } from "./scope.types";
+import type { ScopeHandler } from "./scopeHandler.types";
+import type { ScopeHandlerFactory } from "./ScopeHandlerFactory";
+
+export class BoundedNonWhitespaceSequenceScopeHandler extends BaseScopeHandler {
+  public readonly scopeType = { type: "boundedNonWhitespaceSequence" } as const;
+  public readonly iterationScopeType: ScopeType = {
+    type: "line",
+  } as const;
+  protected readonly isHierarchical = true;
+
+  constructor(
+    private scopeHandlerFactory: ScopeHandlerFactory,
+    _scopeType: ScopeType,
+    private languageId: string,
+  ) {
+    super();
+  }
+
+  *generateScopeCandidates(
+    editor: TextEditor,
+    position: Position,
+    direction: Direction,
+  ): Iterable<TargetScope> {
+    const paintScopeHandler = this.scopeHandlerFactory.create(
+      {
+        type: "nonWhitespaceSequence",
+      },
+      this.languageId,
+    )!;
```

Comment:
I believe if we change this to `block` we get https://github.com/cursorless-dev/cursorless/issues/903 for free; could prob just use a shared base class

Reply:
With the implementation using `oneOf` it's not that much code to add short block in a similar matter.
https://github.com/cursorless-dev/cursorless/blob/9bf56f2053cf062391a27b5279fc6d5a524570ba/packages/cursorless-engine/src/processTargets/modifiers/scopeHandlers/BoundedNonWhitespaceSequenceScopeHandler.ts

Reply:
We now support short block